hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/NameNodeProxiesClient.java

```diff
@@ -349,6 +349,18 @@ public static ClientProtocol createProxyWithAlignmentContext(
       boolean withRetries, AtomicBoolean fallbackToSimpleAuth,
       AlignmentContext alignmentContext)
       throws IOException {
+    if (!conf.getBoolean(HdfsClientConfigKeys.DFS_OBSERVER_READ_ENABLE,
```

Comment:
I have an idea on how to disable Observer Read.
- Proxy always with AlignmentContext, like ClientGSIContext or RouterGSIContext
- Always update the lastSeenStateId from active
- ClientGSIContext has a boolean flag, like disableOberverRead
- If disableObserverRead is true, getLastSeenStateId() will always return a special number, like -100
- And if we want dynamically enable ObserverRead for one ns, we just need to set the disableObserverRead flag to false.

Our cluster with 5w+ DNs used this solution, and it is convenient to control ObserverRead. 
@simbadzina How do you think about this idea?

Reply:
I'll add commented on your draft RB https://github.com/apache/hadoop/pull/4441